### ISSUE TITLE
Issue in TCP collection process when there is an error while decoding packet

### ISSUE THREAD

**Issue by srikartati** [vmware/go-ipfix#124]
**Describe the bug**
https://github.com/vmware-tanzu/antrea/issues/1747#issuecomment-760646600

Antrea Flow Aggregator utilizes the collection process from go-ipfix lib. There was a issue reported there citing crashes flow aggregator application. Following backtraces shows that issue is in TCP collecting process: https://github.com/vmware/go-ipfix/blob/release-0.4.0/pkg/collector/tcp.go#L66

However the reason for panic error is not known.

```
I0112 13:52:00.592685       1 flow-aggregator.go:27] Flow aggregator starting...
I0112 13:52:00.597287       1 tcp.go:38] Start tcp collecting process on [::]:4739
E0112 14:00:55.171757       1 tcp.go:90] collector only supports IPFIX (v10); invalid version 2923 received
panic: runtime error: slice bounds out of range [:232] with capacity 95

goroutine 53 [running]:
github.com/vmware/go-ipfix/pkg/collector.(*CollectingProcess).handleTCPClient.func1(0x91b940, 0xc000107258, 0xc0002d2000, 0xc0019812c0, 0xf)
	/go/pkg/mod/github.com/vmware/go-ipfix@v0.4.2/pkg/collector/tcp.go:88 +0x68a
created by github.com/vmware/go-ipfix/pkg/collector.(*CollectingProcess).handleTCPClient
	/go/pkg/mod/github.com/vmware/go-ipfix@v0.4.2/pkg/collector/tcp.go:65 +0x18a
```

```
E0113 04:30:57.224860       1 process.go:195] Error when closing connection to collector: close udp 10.20.0.248:51440->10.103.98.74:4739: use of closed network connection
panic: runtime error: slice bounds out of range [:249] with capacity 65

goroutine 46 [running]:
github.com/vmware/go-ipfix/pkg/collector.(*CollectingProcess).handleTCPClient.func1(0x91b940, 0xc0171fe000, 0xc000266000, 0xc019a7a010, 0xe)
	/go/pkg/mod/github.com/vmware/go-ipfix@v0.4.2/pkg/collector/tcp.go:88 +0x68a
created by github.com/vmware/go-ipfix/pkg/collector.(*CollectingProcess).handleTCPClient
	/go/pkg/mod/github.com/vmware/go-ipfix@v0.4.2/pkg/collector/tcp.go:65 +0x18a
```

**To Reproduce**
Run Flow aggregator application for a long time.
Figure out better repro steps if possible.

**Expected behavior**
There should not be any crash.

**Actual Behavior**
Flow aggregator application in Antrea should not crash.

**Versions**
Please provide the version of go-ipfix where this problem is observed.

**Additional context**

**Comment by srikartati** [comment]
@zyiou Could you take a shot at this? 
I will try to fix Issue #123 

Let us get these fixes in soon, so Antrea 0.13.0 can consume them.

**Comment by zyiou** [comment]
> @zyiou Could you take a shot at this?
> I will try to fix Issue #123
> 
> Let us get these fixes in soon, so Antrea 0.13.0 can consume them.

Sure. will take a look at collector side

**Comment by srikartati** [comment]
Closing this issue. It is fixed in release-0.4.0 branch. Fix will be cherrypicked onto the master soon.
